frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/delete-block-pool-modal.tsx

```diff
@@ -0,0 +1,154 @@
+        return scList;
+      }, []);
+      const pvcScNames: string[] = pvcResources.items?.map(getStorageClassName);
+
+      // intersection of scNames and pvcScNames
+      const usedScNames = poolScNames.filter((scName) => pvcScNames.includes(scName));
+
+      if (usedScNames.length) {
+        dispatch({ type: BlockPoolActionType.SET_POOL_STATUS, payload: POOL_PROGRESS.BOUNDED });
+        setScNames(commaSeparatedString(usedScNames, t));
+      }
+    }
+  }, [scResources, scLoaded, pvcResources, pvcLoaded, state.poolStatus, poolName, t]);
+
+  // Delete block pool
+  const onClick = () => {
+    handlePromise(k8sKill(CephBlockPoolModel, blockPoolConfig), () => close());
+  };
+
+  const MODAL_TITLE = t('ceph-storage-plugin~Delete Block Pool');
+
+  return (
+    <div className="modal-content modal-content--no-inner-scroll">
+      <ModalTitle close={close}>
+        <YellowExclamationTriangleIcon className="co-icon-space-r" /> {MODAL_TITLE}
+      </ModalTitle>
+      {isLoaded && pvcLoaded && scLoaded && !(loadError && pvcLoadError && scLoadError) ? (
+        <>
+          <ModalBody>
+            {state.poolStatus === POOL_PROGRESS.NOTALLOWED ? (
+              <div key="progress-modal">
+                <BlockPoolStatus
+                  status={state.poolStatus}
+                  name={state.poolName}
+                  error={state.errorMessage}
+                />
+              </div>
+            ) : state.poolStatus === POOL_PROGRESS.BOUNDED ? (
+              <Trans t={t} ns="ceph-storage-plugin">
+                <p>
+                  <strong>{{ poolName }}</strong> cannot be deleted. When a pool is bounded to PVC
+                  it cannot be deleted. Please detach all the resources from storage class(es):{' '}
+                  <strong>{{ scNames }}</strong>.{' '}
+                </p>
+              </Trans>
+            ) : (
+              <Trans t={t} ns="ceph-storage-plugin">
+                <p>
+                  Deleting <strong>{{ poolName }}</strong> will remove all the saved data of this
+                  pool. Are you sure want to delete?
+                </p>
+              </Trans>
+            )}
+          </ModalBody>
+          <ModalFooter inProgress={inProgress}>
+            <BlockPoolModalFooter
+              state={state}
+              dispatch={dispatch}
+              onSubmit={onClick}
+              cancel={cancel}
+              close={close}
+              footerPrimaryAction={FooterPrimaryActions.DELETE}
+            />
+          </ModalFooter>
+        </>
+      ) : (
+        <StatusBox
+          loaded={isLoaded && pvcLoaded && scLoaded}
+          loadError={loadError ?? pvcLoadError ?? scLoadError}
+          label={t('ceph-storage-plugin~Block Pool Delete Modal')}
+        />
+      )}
+    </div>
+  );
+});
+
+type DeleteBlockPoolModalProps = {
+  kind?: string;
+  blockPoolConfig?: StoragePoolKind;
```

Comment:
why all props optional ?

Reply:
And I cant find where `kind` is used ?

Reply:
Kind is optional as per extension design we have to receive both, but kind is optional